### PR TITLE
Fix: highlight sidebar's active section

### PR DESCRIPTION
**Pull request by ixrock**
- fix: highlight active sidebar's section (regression from #2166)
- fix: caching crd definitions to prevent sidebar's scroll jumps
- refactored & simplified `Sidebar` and `SidebarItem`

_Before:_

https://user-images.githubusercontent.com/6377066/111791141-9c7b5880-88cb-11eb-901a-bfb925ca5b0e.mov

_After:_

https://user-images.githubusercontent.com/6377066/111791121-95ece100-88cb-11eb-9ba2-e6e4735712f4.mov

### CODE REVIEW

File: src/renderer/components/layout/sidebar.tsx
Comment (Nokel81) [inline src/renderer/components/layout/sidebar.tsx]: I don't think that we should display the spinner if loading has failed.

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [inline src/renderer/components/layout/sidebar-item.tsx]: Why do we need this change? Isn't the problem that `isActive` wasn't computed?

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [inline src/renderer/components/layout/sidebar-item.tsx]: Changing this has nothing to do with fixing the bug.

**Review by Nokel81 — CHANGES_REQUESTED** [review]
^^^

File: src/renderer/components/layout/sidebar-item.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx]: It's not related to `isActive`. `props.subMenu` has basically same role as `props.children`, could be even removed (currently both adds data to sub-menu container).

File: src/renderer/components/layout/sidebar-item.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx]: And what? I don't calculate changed lines (just re-sorted a bit for better look)

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [reply src/renderer/components/layout/sidebar-item.tsx]: Yeah I have noticed that they are very similar indeed. But for a bug fix I think it should be a small as possible (within reason).

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [reply src/renderer/components/layout/sidebar-item.tsx]: You changed a prop field name, that doesn't have anything to do with this bug fix.

File: src/renderer/components/layout/sidebar.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar.tsx]: I'd prefer to keep it as is since it fixes a bug (see video1) with reloading CRD definitions each time (otherwise more files needs to be tweaked)

File: src/renderer/components/layout/sidebar-item.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx]: Yes, it fits better imho + it's used currently only in `Sidebar` and `SidebarItem`. So it's a good time to change if we want it.

File: src/renderer/components/layout/sidebar-item.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx]: Renamed back to `props.subMenus`

File: src/renderer/components/layout/sidebar.tsx
Comment (Nokel81) [reply src/renderer/components/layout/sidebar.tsx]: Where in the video?

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [inline src/renderer/components/layout/sidebar-item.tsx]:
```suggestion
  /**
   * Forces this item to be also show as active or not.
   *
   * Default: dynamically checks the location against the `url` props to determine if
   * this item should be shown as active
   */
  isActive?: boolean;
```

File: src/renderer/components/layout/sidebar-item.tsx
Comment (Nokel81) [inline src/renderer/components/layout/sidebar-item.tsx:88]: Shouldn't it be possible to just remove this line and fix the bug?

File: src/renderer/components/layout/sidebar.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar.tsx]: When I'm clicking second time at `Definitions` you can see loading indicator and jumping scroll for a short time.

File: src/renderer/components/layout/sidebar-item.tsx
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx:88]: Dunno. CDS's and 3rd level menus are handled a bit differently.

**Comment by ixrock** [comment]
@Nokel81 ready for re-review, PTAL.

**Review by aleksfront — APPROVED** [review]
LGTM

**Review by Nokel81 — APPROVED** [review]
Looks good, can you do a follow up PR to remove `submenu`? So that they are just `children`?

**Comment by ixrock** [comment]
> Looks good, can you do a follow up PR to remove `submenu`? So that they are just `children`?

Yes!

**Comment by Nokel81** [comment]
Next time please don't merge as it make my work with releases slightly more complicated (this is really only a problem when an active release is in progress). Thanks.

**Comment by ixrock** [comment]
> Next time please don't merge as it make my work with releases slightly more complicated (this is really only a problem when an active release is in progress). Thanks.

Alright! Usually you say it when you need to wait some time before merge (for various reasons).

**Comment by Nokel81** [comment]
True, sorry.

**Pull request merged**
